lib/transport/tls-context.c

```diff
@@ -593,8 +595,10 @@ tls_context_setup_context(TLSContext *self)
   if (!tls_context_setup_sigalgs(self))
     goto error;
 
+#if SYSLOG_NG_HAVE_DECL_SSL_CONF_CTX_new
   if (!tls_context_setup_cmd_context(self))
     goto error;
+#endif
```

Comment:
we should report an error if the user has used the command list feature (e.g. openssl_conf_cmds in the afsocket drivers), because this way those commands are basically ignored.

So I would leave the call here and display an error in tls_context_setup_cmd_context() if conf_cmds_list is non-NULL and we are building with libressl.

Otherwise looks ok.

Reply:
I think we should follow the error reporting pattern that we already have here:
https://github.com/syslog-ng/syslog-ng/blob/ede106afd7d70a8f962acf10a4e3f4a50a70cb15/lib/transport/tls-context.c#L841-L852

`tls_context_set_conf_cmds()` already has a `GError` parameter, we should just use it.
https://github.com/syslog-ng/syslog-ng/blob/ede106afd7d70a8f962acf10a4e3f4a50a70cb15/lib/transport/tls-context.c#L856C1-L856C26

Reply:
Updated as requested